### PR TITLE
[helpers TS conversion] subclasses

### PR DESCRIPTION
**Pull request by amjed-98**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes part of #16500 
| Patch: Bug Fix?          | No
| Major: Breaking Change?  | No
| Minor: New Feature?      | No
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  | No
| License                  | MIT

Migrate subclasses helpers to typescript (callSuper, inherits, inheritsLoose, isNativeFunction, wrapNativeSuper)

### CODE REVIEW

File: packages/babel-helpers/src/helpers/isNativeFunction.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
 /* @minVersion 7.0.0-beta.0 */
 
-export default function _isNativeFunction(fn) {
+export default function _isNativeFunction(fn: unknown): fn is Function {
   // Note: This function returns "true" for core-js functions.
   try {
-    return Function.toString.call(fn).indexOf("[native code]") !== -1;
-  } catch (e) {
+    return Function.toString.call(fn).includes("[native code]");
```
Comment (liuxingbaoyu) [inline packages/babel-helpers/src/helpers/isNativeFunction.ts:6]: Can you undo this change? Because `includes` are not supported in older engines.

Comment (amjed-98) [reply packages/babel-helpers/src/helpers/isNativeFunction.ts:6]: yes I will now

the reason I changed it is because it was giving me an error on ci pipline

**Review by nicolo-ribaudo — APPROVED** [review]
Thanks!

File: packages/babel-helpers/src/helpers/wrapNativeSuper.ts [hunk 2]
```diff
@@ -19,7 +20,9 @@ export default function _wrapNativeSuper(Class) {
       if (_cache.has(Class)) return _cache.get(Class);
       _cache.set(Class, Wrapper);
     }
+
     function Wrapper() {
+      // @ts-expect-error -- where are sure Class is a function here
```
Comment (JLHwung) [inline packages/babel-helpers/src/helpers/wrapNativeSuper.ts:25]:
```suggestion
      // @ts-expect-error -- we are sure Class is a function here
```

File: packages/babel-helpers/src/helpers/wrapNativeSuper.ts [hunk 3]
```diff
@@ -2,15 +2,16 @@
...
+import setPrototypeOf from "./setPrototypeOf.ts";
+import isNativeFunction from "./isNativeFunction.ts";
+import construct from "./construct.ts";
 
-export default function _wrapNativeSuper(Class) {
+export default function _wrapNativeSuper(Class: Function | null) {
   var _cache = typeof Map === "function" ? new Map() : undefined;
 
-  _wrapNativeSuper = function _wrapNativeSuper(Class) {
+  // @ts-expect-error -- test
```
Comment (JLHwung) [inline packages/babel-helpers/src/helpers/wrapNativeSuper.ts:13]:
```suggestion
  // @ts-expect-error -- reuse function id for helper size
```

**Review by JLHwung — APPROVED** [review]
Thanks.

**Review by liuxingbaoyu — APPROVED** [review]
Thanks!
